Yoast/javascript
# SnippetPreview: Change the desktop mode URL

## Summary


This PR can be summarized in the following changelog entry:

* [@yoast/search-metadata-previews] Improves the URL in the SnippetEditor.
* [Yoast SEO] Improves the URL in the Google preview when in the desktop mode.

## Relevant technical choices:

* Move URL to the top
* Change styling to match current Google style:
  * Base + rest using different colors
* Fix URL parsing by removing the `http(s)://` stripping in the SnippetEditor
* Always strip the protocol, `http` and `https`

## Test instructions

This PR can be tested by following these steps:

* Edit a post
* Open the Google preview
* Switch it to the desktop mode
* Ensure you have a description (the actual description has a different color than the placeholder)
* Do a google search on a desktop and verify it looks similar regarding the URL
* The URL is now split in base + rest, each has a different color
* The URL size changed
* The URL arrow changed color and position (moved to the top along with the URL)
* The desktop description color changed

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  * The base URL had to be fixed from the SnippetEditor
  * Make sure the mobile view still looks the same
  * Make sure the analysis still catches the keyphrase in the slug

## UI changes

Before:
![before](https://user-images.githubusercontent.com/35524806/109526462-de766300-7ab2-11eb-97d4-75c7602851b8.png)
After:
![after](https://user-images.githubusercontent.com/35524806/109526806-4462ea80-7ab3-11eb-9d91-8f14318a698c.png)

## Quality assurance


Fixes https://yoast.atlassian.net/browse/P1-60
